homeassistant/components/modbus/__init__.py

```diff
@@ -178,73 +146,136 @@ def write_coil(service):
         client_name = service.data[ATTR_HUB]
         hub_collect[client_name].write_coil(unit, address, state)
 
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, start_modbus)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, start_modbus)
 
     return True
 
 
 class ModbusHub:
-    """Thread safe wrapper class for pymodbus."""
+    """Wrapper class for pymodbus."""
 
-    def __init__(self, modbus_client, name):
+    def __init__(self, client_config, main_loop):
         """Initialize the Modbus hub."""
-        self._client = modbus_client
-        self._lock = threading.Lock()
```

Comment:
It looks like the lock was removed.  I think it needs to be converted to a `asyncio.Lock()` instead.

Reply:
Done (will be uploaded when all changes are ready). I removed thread.lock() because I had a deadlock, and simply forgot to add async.Lock().